hugr-core/src/hugr/serialize.rs

```diff
@@ -215,7 +215,8 @@ impl TryFrom<&Hugr> for SerHugrLatest {
             let op = hugr.get_optype(node);
             let is_value_port = offset < op.value_port_count(dir);
             let is_static_input = op.static_port(dir).is_some_and(|p| p.index() == offset);
-            let offset = (is_value_port || is_static_input).then_some(offset as u32);
+            let is_cfg_edge = op.is_dataflow_block();
+            let offset = (is_value_port || is_static_input || is_cfg_edge).then_some(offset as u32);
```

Comment:
This is too restraining.
The bug any time we have more than one non-df ports, so it would be safer to do

```suggestion
            let multiple_other_ports = op.non_df_port_count(dir) > 1;
            let offset = (is_value_port || is_static_input || multiple_other_ports).then_some(offset as u32);
```

Alternatively, encode the offset always 🤷 

It's not a breaking serialization change either way since the decoding logic works fine if the offset is already defined

Reply:
Ah, seems like `hugr-py` uses `None` only for order edges.

If we wanted to follow that here then we can check
```rust
let other_port_is_not_order = op.other_port_kind(dir) != EdgeKind::StateOrder
let offset = (is_value_port || is_static_input || other_port_is_not_order).then_some(offset as u32);
```